### PR TITLE
[inductor] Refactor IR printing

### PR DESCRIPTION
**Pull request by jansel**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #96024
* #96023

Reland #95567 part 2.  The previous version of this had a bug which that
added test triggers.

cc @soumith @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @desertfire

### CODE REVIEW

File: torch/_inductor/virtualized.py [hunk 1]
```diff
@@ -96,9 +97,35 @@ def inner(*args):
...
+        args = [index, rindex] if rindex is not None else [index]
+        names = ["index", "rindex"] if rindex is not None else ["index"]
+        formatter = KernelFormatterHandler(MockHandler())
+
+        with formatter.output.indent(-1):
+            formatter.output.writeline(f"def inner_fn({', '.join(names)}):")
+        for name, arg in zip(names, args):
+            if arg:
+                lhs = ", ".join(
+                    [
```
Comment (Skylion007) [inline torch/_inductor/virtualized.py:116]: This can be a generator instead of a list comprehension

Comment (jansel) [reply torch/_inductor/virtualized.py:116]: I don't think it matters since this list will be <10 elements.

**Review by ngimel — APPROVED** [review]

**Comment by ngimel** [comment]
cc @desertfire, somehow we didn't catch this memory compression regression in CI, perhaps makes sense to also add SuperSloMo (if it's memory compression is stable enough)

**Comment by jansel** [comment]
> @jansel I bisected a memory compression regression on Super_SloMo (0.967 -> 0.8615) to this PR, can you take a look?

Weird.  Perhaps we need to increase `realize_bytes_threshold` because the strings are slightly longer now.

**Pull request closed without merging**